components/base/kinematicbase/ptgKinematics.go

```diff
@@ -157,21 +159,28 @@ func (ptgk *ptgBaseKinematics) GoToInputs(ctx context.Context, inputs []referenc
 		linVel := r3.Vector{0, trajNode.LinVelMMPS, 0}
 		angVel := r3.Vector{0, 0, rdkutils.RadToDeg(trajNode.AngVelRPS)}
 
-		ptgk.logger.Debugf(
-			"setting velocity to linear %v angular %v and running velocity step for %s",
-			linVel,
-			angVel,
-			timestep,
-		)
-
-		err := ptgk.Base.SetVelocity(
-			ctx,
-			linVel,
-			angVel,
-			nil,
-		)
-		if err != nil {
-			return multierr.Combine(err, ptgk.Base.Stop(ctx, nil))
+		// This should call SetVelocity if:
+		// 1) this is the first iteration of the loop, or
+		// 2) either of the linear or angular velocities has changed
+		if i == 0 || !(linVel.ApproxEqual(lastLinVel) && angVel.ApproxEqual(lastAngVel)) {
+			ptgk.logger.Debugf(
+				"setting velocity to linear %v angular %v and running velocity step for %s",
+				linVel,
+				angVel,
+				timestep,
+			)
+
+			err := ptgk.Base.SetVelocity(
```

Comment:
could bring this into the conditional below

Reply:
Stylistically I personally just don't like the 
```
if x := func(); cond(x) {
   stuff
}
```
paradigm as much as keeping it separate. Easier to tell where the scopes are that way.